test/unit/NodeCompiler/CompileNodeToValueTest.php

```diff
@@ -251,4 +251,37 @@ public function method($param = static::BAR) {}
 
         $this->assertSame('baz', $paramInfo->getDefaultValue());
     }
+
+    public function testClassConstantClassNameResolution()
+    {
+        $phpCode = '<?php
+
+        class Foo {
+        }
+        class Bat {
+            const QUX = Foo::class;
+        }
+        ';
+
+        $reflector = new ClassReflector(new StringSourceLocator($phpCode));
+        $classInfo = $reflector->reflect('Bat');
+        $this->assertSame('Foo', $classInfo->getConstant('QUX'));
+    }
+
+    public function testClassConstantClassNameAliasResolution()
+    {
+        $phpCode = '<?php
+        namespace Bar;
+
+        class Foo {
+        }
+        class Bat {
+            const QUX = Foo::class;
+        }
+        ';
+
+        $reflector = new ClassReflector(new StringSourceLocator($phpCode));
+        $classInfo = $reflector->reflect('Bat');
```

Comment:
@Ocramius the other test should have been `Bar\Bat` this one isn't namespaced. Fixed that now.